packages/kbn-unified-field-list/src/services/field_existing/field_existing_utils.ts

```diff
@@ -55,9 +55,13 @@ export async function fetchFieldExistence({
     pattern: '',
     indexFilter: toQuery(timeFieldName, fromDate, toDate, dslQuery),
   });
+  const hasNewFields = existingFieldList.some(
+    (field) => dataView.getFieldByName(field.name) === undefined
+  );
```

Comment:
@elastic/kibana-data-discovery here's where the request for the available fields of the UnifiedFieldList is detecting new fields incoming. This triggers a callback setting a state of the field list to show UI to reload the data view + refetch the query. ideally just the data view needs to be reloaded, something we could implement in the near future. Anyhow, once caching is in place, reloading the data view needs to clean the cache of the data view's fields